main.py

```diff
@@ -35,12 +38,23 @@ def main(
     :type superusers: str
     :type lockFolder: str
     :type unlockFolder: str
```

Comment:
I still need to think about this change a bit more, however, I would like to have the main function just for convenience and encourage developers/studio's to call their custom library widget directly.  We should probably make this change a separate pull request too.

Example:

```python
import customlibrarywidget
w = customlibrarywidget.CustomLibraryWidget.instance()
```

It would also be nice to avoid a "studiolibrarymaya" dependency in the studiolibrary code.

I'll need to think about this more, however, I was thinking something like the following.

```python
def main(cls=None, **kwargs):
    """
    Convenience method for creating a libraryWidget instance.
    """
    cls = cls or studiolibrary.LIBRARY_WIDGET_CLASS or studiolibrary.LibraryWidget
    return cls.instance(**kwargs)
```

To support the current main function we would still need to do something like the following for now.

```python
#studiolibrary/config.py

if studiolibrary.isMaya():
    import studiolibrarymaya.mayalibrarywidget
    studiolibrary.LIBRARY_WIDGET_CLASS = studiolibrarymaya.mayalibrarywidget.MayaLibraryWidget
```

Happy to discuss this further. :)